### ISSUE TITLE
Problem messages can get offset by frequent changes to the document

### ISSUE THREAD

**Issue by 0dinD** [redhat-developer/vscode-java#1633]
When editing a Java file with problems in it, the underlines (reported problem positions) can sometimes become offset. Usually, this fixes itself after a second or so of delay, but in some rare cases the problem underline gets stuck in the wrong position until further changes are made to the document.

##### Environment
- Visual Studio Code version: 1.49.1
- Java extension version: 0.67.0

##### Steps To Reproduce
1. Open a Java class with problems in it (unused variable warnings, for example).
2. Make some changes in quick succession.
3. You should see the error messages jumping around and getting offset to incorrect positions.
4. In some rare cases, the error messages get stuck on the wrong location until a new character is typed.

##### Current Result
![problem-offset-bug-1](https://user-images.githubusercontent.com/52179873/93882362-6b391b80-fce0-11ea-9b33-df6f21509f70.png)
![problem-offset-bug-2](https://user-images.githubusercontent.com/52179873/93882376-6f653900-fce0-11ea-8499-56e39b87df1c.png)

##### Expected Result
The error messages (underlines) should always appear on the correct positions, and never get stuck on the wrong ones, like in the screenshot above.

##### Additional Informations
As you can see in the screenshots above, not only are the error messages offset, the semantic tokens are also offset in the exact same way (see #1597). I've submitted two PRs (#1632, eclipse/eclipse.jdt.ls#1552) to fix the semantic highlighting issue, but this issue with problem messages still persists since I only changed the semantic highlighting implementation. From what I can tell in the screenshots above, the two issues seem highly related, so a fix for this PR can probably be made by looking at how I fixed #1597.

This is how the semantic tokens look with #1632 and eclipse/eclipse.jdt.ls#1552 applied, I would expect the problem underlines to behave in the same way:
![problem-offset-bug-3](https://user-images.githubusercontent.com/52179873/93885574-932a7e00-fce4-11ea-8cb7-6e2d223b3170.png)

**Comment by 0dinD** [comment]
If anyone wants a more detailed showcase on how to reproduce the issue, [see my comment here](https://github.com/eclipse/eclipse.jdt.ls/pull/1552#issuecomment-698345369). Those instructions apply to this issue as well.

**Comment by 0dinD** [comment]
Here's another example, which occurred when applying a fix from the integrated merge conflict resolver in VS Code. The errors were caused by the conflict markers, but when the document was changed with the conflict resolver, the errors became offset to the wrong location, and also didn't go away until the document was changed again. I'm guessing the cause is that the conflict resolver is making multiple document edits in quick succession when applying the fix.

![offset-bug-error](https://user-images.githubusercontent.com/52179873/94724283-57ce2600-035a-11eb-9d3b-4f153ccfec2b.png)

This is what I mean by "conflict resolver". The fix I applied was "Accept Incoming Change".

![conflict-resolver](https://user-images.githubusercontent.com/52179873/94725116-91536100-035b-11eb-95b2-786a512eb510.png)
